GVFS/GVFS.Service/Mac/GVFSService.cs

```diff
@@ -61,12 +88,78 @@ private void ServiceThreadMain()
             }
         }
 
+        private void HandleRequest(ITracer tracer, string request, NamedPipeServer.Connection connection)

```

Comment:
This method is same as the one on `GVFS.Service.Window` *minus* `EnableAndAttachProjFSRequest`

Reply:
It would be really nice if we could find a way to share the code in all of these cases, even if we need a method that only windows has an implementation for so that we don't duplicate so much.  My worry is when we change things, if we don't remember to change it in both places, the code base diverges and is harder to maintain.

Reply:
- Created new `RequestHandler` class to handle `Service` requests.
- Windows handles additional `EnableAndAttachProjFSRequest` requests in a `WindowsRequestHandler` subclass.